File: src/main/java/com/ocds/tender/config/ServiceConfig.java
package com.ocds.tender.config;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.ocds.tender.config.properties.OCDSProperties;
import com.ocds.tender.utils.JsonUtil;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

@Configuration
@ComponentScan(basePackages = "com.ocds.tender.service")
public class ServiceConfig {

    @Bean
    @Primary
    public ObjectMapper objectMapper() {
        ObjectMapper jackson2ObjectMapper = new ObjectMapper();
        jackson2ObjectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
        return jackson2ObjectMapper;
    }

    @Bean
    public JsonUtil jsonUtil(ObjectMapper objectMapper){
        return new JsonUtil(objectMapper);
    }
}

File: src/main/kotlin/com/procurement/access/config/ServiceConfig.kt
package com.procurement.access.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.procurement.access.config.properties.OCDSProperties
import com.procurement.access.service.validation.JsonValidationService
import com.procurement.access.service.validation.MedeiaValidationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@Configuration
@ComponentScan(
    basePackages = [
        "com.procurement.access.service",
        "com.procurement.access.application.service",
        "com.procurement.access.infrastructure.handler"
    ]
)
@EnableConfigurationProperties(value = [OCDSProperties::class])
class ServiceConfig {

    @Autowired
    private lateinit var objectMapper: ObjectMapper

    @Bean
    fun jsonValidationService(): JsonValidationService = MedeiaValidationService(objectMapper)
}
